system_modules/napaudio/src/audio/utility/audiofileutils.cpp

```diff
@@ -159,12 +164,71 @@ namespace nap
 		
 		bool readAudioFile(const std::string& fileName, MultiSampleBuffer& output, float& outSampleRate, nap::utility::ErrorState& errorState)
 		{
-			if (utility::toLower(utility::getFileExtension(fileName)) == "mp3")
-				return readMp3File(fileName, output, outSampleRate, errorState);
+			// if (utility::toLower(utility::getFileExtension(fileName)) == "mp3")
+			// 	return readMp3File(fileName, output, outSampleRate, errorState);
 			
 			return readLibSndFile(fileName, output, outSampleRate, errorState);
 		}
 		
 		
+		//this function should probably live in nap::math
+		bool isFloatEqual(const float & a, const float & b) {
+			return std::abs(a - b) <= std::numeric_limits<float>::epsilon() * std::abs(a);
+		}
+
+
+		bool NAPAPI resampleSampleBuffer(MultiSampleBuffer& buffer, float sourceSampleRate, float destSampleRate, int converterType, nap::utility::ErrorState& errorState){
```

Comment:
`convertType` should be the `EResampleType` -> type safe. Consider moving the `EResampleType` to `audiofileutils`